metadata_mapper/validate_mapping.py

```diff
@@ -18,330 +29,143 @@ def solr(**params):
-def validate_mapped_collection(payload):
-    payload = json.loads(payload)
-    collection_id = payload.get('collection_id')
-
-    if settings.DATA_SRC == 'local':
-        mapped_path = settings.local_path('mapped_metadata', collection_id)
-        page_list = [f for f in os.listdir(mapped_path)
-                     if os.path.isfile(os.path.join(mapped_path, f))]
-
-    solr_records = {}
+def get_solr_data(collection_id: int, ids: list[str]) -> list[dict]:
+    collection_url = ("collection_url:\"https://registry.cdlib.org/api/v1/"
+                      f"collection/{collection_id}/\"")
+    harvest_id_q = " OR ".join([f"harvest_id_s:\"{f}\"" for f in ids])
     query = {
-        'fq': (
-            "collection_url: \"https://registry.cdlib.org/api/v1"
-            f"/collection/{collection_id}/\""
-        ),
-        'rows': 100,
+        'fq': f"{collection_url} AND ({harvest_id_q})",
+        'rows': len(ids),
         'start': 0
     }
-    collection_report = ["severity, type, couch id, field_name, rikolti value, solr value"]
-
-    for page in page_list:
-        if settings.DATA_SRC == 'local':
-            page_path = os.sep.join([
-                settings.local_path('mapped_metadata', collection_id),
-                str(page)
-            ])
-            page = open(page_path, "r")
-            mapped_metadata = page.read()
-        elif settings.DATA_SRC == 's3':
-            s3 = boto3.resource('s3')
-            bucket = 'rikolti'
-            key = f"mapped_metadata/{collection_id}/{page}"
-            s3_obj_summary = s3.Object(bucket, key).get()
-            mapped_metadata = s3_obj_summary['Body'].read()
-
-        rikolti_records = json.loads(mapped_metadata)
-        rikolti_records = {
-            f"{collection_id}--{r['calisphere-id']}": r
-            for r in rikolti_records
-        }
-
-        print(f"[{collection_id}]: Validating page {page_path.split('/')[-1]}")
-        query, solr_records, page_report = validate_mapped_page(
-            rikolti_records, solr_records, query)
-        collection_report = collection_report + page_report
+
+    response = solr(**query)
+    return response.get("response", {"docs": None}).get("docs", [])
+
+
+def get_validator_class(collection_id: int) -> Type[Validator]:
+    url = ("https://registry.cdlib.org/api/v1/rikoltimapper/"
+           f"{collection_id}/?format=json")
+
+    try:
+        response = requests.get(url=url)
+        response.raise_for_status()
+        collection_data = response.json()
+    except requests.exceptions.HTTPError as err:
         print(
-            f"[{collection_id}]: Validated page {page_path.split('/')[-1]} "
-            f"- {len(collection_report)} errors"
+            f"[Collection {collection_id}]: "
+            f"[{url}]"
+            f"{err}; A valid collection id is required for validation"
         )
+        return
+
+    mapper = collection_data.get("rikolti_mapper_type")
+    mapped_mapper = MAPPER_MAP.get(mapper, mapper)
```

Comment:
Shouldn't need this? `rikolti_mapper_type` is already in the format of `oai.islandora`, for example.

Reply:
Yeah, not anymore. I'd imagine this should be (or has been?) addressed elsewhere, then we can rebase on that.